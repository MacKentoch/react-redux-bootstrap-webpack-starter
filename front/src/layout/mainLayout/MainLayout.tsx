import React, { Fragment, useEffect, useCallback, ReactNode } from 'react';
import { withRouter } from 'react-router';
import { RouteComponentProps } from 'react-router';
import NavigationBar from '../../components/navigation';
import BackToTop from '../../components/backToTop/BackToTop';
import navigationModel from '../../config/navigation';
import registerServiceWorker from '../../services/sw/registerServiceWorker';

type Props = {
  children: ReactNode;
} & RouteComponentProps;

function MainLayout({ children }: Props) {
  // #region on mount effect
  useEffect(() => {
    if (typeof window !== undefined) {
      // register service worker (no worry about multiple attempts to register, browser will ignore when already registered)
      registerServiceWorker();
    }
  }, []);
  // #endregion

  // #region callbacks
  /* eslint-disable no-unused-vars*/
  const handleLeftNavItemClick = useCallback(
    (/* event: React.MouseEvent<HTMLButtonElement>, viewName?: string */) => {
      // something to do here?
    },
    [],
  );

  const handleRightNavItemClick = useCallback(
    (/* event: React.MouseEvent<HTMLButtonElement>, viewName?: string */) => {
      // something to do here?
    },
    [],
  );
  // #endregion

  return (
    <Fragment>
      <div id="appContainer">
        <NavigationBar
          brand={navigationModel.brand}
          navModel={navigationModel}
          leftNavItemClick={handleLeftNavItemClick}
          rightNavItemClick={handleRightNavItemClick}
        />
        <div className="container-fluid">{children}</div>
        <BackToTop minScrollY={40} scrollTo={'appContainer'} />
      </div>
    </Fragment>
  );
}

export default withRouter(MainLayout);
