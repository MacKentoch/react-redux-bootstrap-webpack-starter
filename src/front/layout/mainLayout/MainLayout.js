// @flow

import React, { Fragment, useEffect, useCallback } from 'react';
import { withRouter } from 'react-router';
import { type Match, type Location, type RouterHistory } from 'react-router';
import NavigationBar from '../../components/navigation';
import BackToTop from '../../components/backToTop/BackToTop';
import navigationModel from '../../config/navigation';
import registerServiceWorker from '../../services/sw/registerServiceWorker';

// #region flow types
type Props = {
  // from withRouter HOC:
  match: Match,
  location: Location,
  history: RouterHistory,

  children: any,
};
// #endregion

// #region MainLayout HOC
function MainLayout({ history, location, match, children }: Props) {
  // #region on mount effect
  useEffect(() => {
    if (typeof window !== undefined) {
      // register service worker (no worry about multiple attempts to register, browser will ignore when already registered)
      registerServiceWorker();
    }
  }, []);
  // #endregion

  // #region callbacks
  /* eslint-disable no-unused-vars*/
  const handleLeftNavItemClick = useCallback((
    event: SyntheticEvent<>,
    viewName: string,
  ) => {
    // something to do here?
  }, []);

  const handleRightNavItemClick = useCallback((
    event: SyntheticEvent<>,
    viewName: string,
  ) => {
    // something to do here?
  }, []);
  // #endregion

  return (
    <Fragment>
      <div id="appContainer">
        <NavigationBar
          brand={navigationModel.brand}
          navModel={navigationModel}
          handleLeftNavItemClick={handleLeftNavItemClick}
          handleRightNavItemClick={handleRightNavItemClick}
        />
        <div className="container-fluid">{children}</div>
        <BackToTop minScrollY={40} scrollTo={'appContainer'} />
      </div>
    </Fragment>
  );
}

// $FlowIgnore
export default withRouter(MainLayout);
