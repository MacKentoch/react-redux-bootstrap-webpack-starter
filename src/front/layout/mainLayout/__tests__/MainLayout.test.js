import React from 'react';
import { MemoryRouter } from 'react-router';
import { shallow } from 'enzyme';
import MainLayout from '../MainLayout';

describe('MainLayout', () => {
  it('renders as expected', () => {
    const component = shallow(
      <MemoryRouter>
        <MainLayout />
      </MemoryRouter>,
    );
    expect(component).toMatchSnapshot();
  });
});
