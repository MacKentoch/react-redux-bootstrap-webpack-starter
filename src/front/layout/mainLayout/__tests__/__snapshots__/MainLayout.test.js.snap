// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MemoryRouter>
    <withRouter(MainLayout) />
  </MemoryRouter>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <withRouter(MainLayout) />,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "le0r2y",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "le0r2y",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <withRouter(MainLayout) />,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "le0r2y",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "le0r2y",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
